server/src/main/java/org/elasticsearch/index/get/ShardGetService.java

```diff
@@ -333,6 +337,19 @@ private GetResult innerGetFetch(
             }
         }
 
+        // NOTE: when _ignored is requested via `stored_fields` we need to load it from doc values instead of loading it from stored fields.
+        // The _ignored field used to be stored, but as a result of supporting aggregations on it, it moved from using a stored field to
+        // using doc values.
+        if (storedFields != null && Arrays.asList(storedFields).contains(IgnoredFieldMapper.NAME)) {
```

Comment:
Don't we need to execute the new logic based on index created version? What happens currently is that we try to load it from both doc values and stored fields regardless of index created version? That works but may not be the best choice? I don't know though what performance impact this has given that it's all done against a memory index.

Reply:
Yes right, I did that later and I forgot to push actually.